devops/scripts/create-sycl-user.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+
+set -e
+
+if [[ $# -eq 0 ]]; then
+  # When launched without arguments, we assume that it was launched as part of
+  # CI workflow and therefore a different kind of user is created
+  USER_NAME=sycl_ci
+  SET_PASSWD=true
+
+  # By default Ubuntu sets an arbitrary UID value, that is different from host
+  # system. When CI passes default UID value of 1001, some of LLVM tools fail to
+  # discover user home directory and fail a few LIT tests. Fixes UID and GID to
+  # 1001, that is used as default by GitHub Actions.
+  USER_ID=1001
+else
+  if [[ "${1:-}" != "--regular" ]]; then
+    echo "The only supported argument is --regular!"
+    exit 1
+  fi
+  USER_NAME=sycl
+  SET_PASSWD=false
+
+  # Some user id which is different from the one assigned to sycl_ci user
+  USER_ID=1234
+fi
+
+groupadd -g $USER_ID $USER_NAME && useradd $USER_NAME -u $USER_ID -g $USER_ID -m -s /bin/bash
+# Add user to video/irc groups so that it can access GPU
+usermod -aG video $USER_NAME
+usermod -aG irc $USER_NAME
+
+# group 109 is required for user to access PVC card.
+groupadd -f -g 109 render
+usermod -aG render $USER_NAME
+
+if [[ $SET_PASSWD == true ]]; then
+  if [[ ! -f /run/secrets/sycl_ci_passwd ]]; then
```

Comment:
Can you remind me from where will `/run/secrets/sycl_ci_passwd` file come from?

Reply:
i had the same question, it looks like its set up by the github runner infrastructure when you have a secret in a workflow and are doing something in docker

Reply:
> Can you remind me from where will `/run/secrets/sycl_ci_passwd` file come from?

It is automounted by `docker` when you pass secrets to a container

Reply:
PS:- I sometimes use Docker containers to debug CI-related issues and I'm wondering if I were to change how the container is created?

Reply:
sorry not sure if i understand your question, but i dont think theres any way to get the secret in that case (i made the secret and i cant even get it anymore, i randomly generated it and once you put it into the github secrets you, you cant get the value of it again lol)

 but if youre calling docker manually, you can log in as any user any time because there's a user argument to `docker exec` so you shouldnt actually need the sudo password, or you can change it to something you know

maybe a bit annoying, but better for security

Reply:
> if youre calling docker manually, you can log in as any user any time because there's a user argument to docker exec so you shouldnt actually need the sudo password, or you can change it to something you know by logging in as root or something

Perfect. Thanks.